codalab/server/bundle_manager.py

```diff
@@ -637,20 +635,33 @@ def _check_resource_failure(
                 return global_fail_string % (pretty_print(value), pretty_print(global_max))
         return None
 
-    def _get_staged_bundles_to_run(self):
+    def _get_matched_workers(self, request_queue, workers):
+        """
+        Get all of the workers that match with the name of the requested worker
+        :param request_queue: the requested worker
+        :param workers: a list of workers
+        :return: a list of matched workers
+        """
+        tagm = re.match('tag=(.+)', request_queue)
+        matched_workers = [worker for worker in workers.workers() if worker['tag'] == tagm.group(1)]
+        if not matched_workers:
+            return []
```

Comment:
Seems like this is not necessary when `matched_workers = []` (again, be explicit about types)

Reply:
This is necessary, as the function will return early so that it won't need to spend time going through the rest of the logic.

Reply:
The rest of the function is just `return matched_workers`, no?